pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -271,8 +280,8 @@ func configFromEnv() (cfg Config, ok bool) {
 	}
 
 	ok = cfg.Global.AuthURL != "" &&
-		cfg.Global.Username != "" &&
-		cfg.Global.Password != "" &&
+		((cfg.Global.Username != "" && cfg.Global.Password != "") ||
+			(cfg.Global.ApplicationCredentialName != "" && cfg.Global.ApplicationCredentialSecret != "")) &&
```

Comment:
Please make this (cfg.Global.ApplicationCredentialID != "" || cfg.Global.ApplicationCredentialName != "") to make it possible to use ID+SECRET in cloud.conf in addition to NAME+SECRET

Reply:
so you want to allow both conditions, right? 

and in addition, seems following should be right based on your ID+SECRET comments? 
(cfg.Global.ApplicationCredentialID != "" || cfg.Global.ApplicationCredentialSecret != "")

Reply:
Well you need to provide SECRET plus one of ID or NAME so I guess it should be something like

(cfg.Global.ApplicationCredentialSecret != "" && (cfg.Global.ApplicationCredentialID != "" || cfg.Global.ApplicationCredentialName != ""))